apps/meteor/server/services/video-conference/service.ts

```diff
@@ -0,0 +1,238 @@
+import { Db } from 'mongodb';
+import type {
+	IRoom,
+	IUser,
+	IVideoConference,
+	MessageTypesValues,
+	VideoConferenceInstructions,
+	DirectCallInstructions,
+	AtLeast,
+	IMessage,
+} from '@rocket.chat/core-typings';
+import { VideoConferenceStatus, isDirectVideoConference } from '@rocket.chat/core-typings';
+
+import { MessagesRaw } from '../../../app/models/server/raw/Messages';
+import { RoomsRaw } from '../../../app/models/server/raw/Rooms';
+import { VideoConferenceRaw } from '../../../app/models/server/raw/VideoConference';
+import { UsersRaw } from '../../../app/models/server/raw/Users';
+import type { IVideoConfService, VideoConferenceJoinOptions } from '../../sdk/types/IVideoConfService';
+import { ServiceClassInternal } from '../../sdk/types/ServiceClass';
+import { settings } from '../../../app/settings/server';
+
+// This class is only here for testing, this code will be handled by apps
+class JitsiApp {
+	static async generateUrl(callId: IVideoConference['_id']): Promise<string> {
+		return `https://jitsi.rocket.chat/${callId}`;
+	}
+
+	static async customizeUrl(
+		call: IVideoConference,
+		user: AtLeast<IUser, '_id' | 'username' | 'name'>,
+		options: VideoConferenceJoinOptions,
+	): Promise<string> {
+		const configs = [
+			`userInfo.displayName="${user.name}"`,
+			`config.prejoinPageEnabled=false`,
+			`config.requireDisplayName=false`,
+			`config.callDisplayName="Direct Message"`,
+		];
+
+		console.log(user.username, options);
+		if (options.mic !== undefined) {
+			configs.push(`config.startWithAudioMuted=${options.mic ? 'false' : 'true'}`);
+		}
+		if (options.cam !== undefined) {
+			configs.push(`config.startWithVideoMuted=${options.cam ? 'false' : 'true'}`);
+		}
+
+		const configHash = configs.join('&');
+		const url = `${call.url}#${configHash}`;
+
+		return url;
+	}
+}
+
+export class VideoConfService extends ServiceClassInternal implements IVideoConfService {
+	protected name = 'video-conference';
+
+	private Messages: MessagesRaw;
+
+	private Users: UsersRaw;
+
+	private Rooms: RoomsRaw;
+
+	private VideoConference: VideoConferenceRaw;
+
+	constructor(db: Db) {
+		super();
+
+		this.Users = new UsersRaw(db.collection('users'));
```

Comment:
nothing wrong here, just something for us to think about, we've been using this pattern in a few places, I wonder if we can came up with a "better" solution to use models inside services.. 

I think just an `enum` of collection names would help.. this is one of my main concerns.